src/prefect/environments/storage/base.py

```diff
@@ -20,14 +20,42 @@ class Storage(metaclass=ABCMeta):
             all flows which utilize this storage class
         - secrets (List[str], optional): a list of Prefect Secrets which will be used to populate `prefect.context`
             for each flow run.  Used primarily for providing authentication credentials.
+        - labels (List[str], optional): a list of labels to associate with this `Storage`.
+        - add_default_labels (bool): If `True`, adds the storage specific default label (if applicable)
+            to the storage labels. Defaults to `True`.
     """
 
-    def __init__(self, result: Result = None, secrets: List[str] = None,) -> None:
+    def __init__(
+        self,
+        result: Result = None,
+        secrets: List[str] = None,
+        labels: List[str] = None,
+        add_default_labels: bool = True,
+    ) -> None:
         self.result = result
         self.secrets = secrets or []
+        self._labels = labels or []
+        self.add_default_labels = add_default_labels
```

Comment:
I'm tempted to think we should expose this as a configuration setting, maybe in [this section](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/config.toml#L117)

Reply:
Good with me. Thoughts on still including as part of the constructor? Or exclusively a configuration option?

Reply:
I actually like it as a part of the constructor (it's a relatively common pattern for us), so something like:
```python
def __init__(self, add_default_labels: bool = None):
    if add_default_labels is None:
        self.add_default_labels = config.flows.storage.add_default_labels
    else:
        self.add_default_labels = add_default_labels
```

Reply:
Gotcha! Without the serialization changes and only being needed prior to registering flows, this makes sense to me!